app/src/features/apiClient/screens/apiClient/components/sidebar/components/collectionsList/requestRow/RequestRow.tsx

```diff
@@ -8,14 +8,37 @@ import { RQButton } from "lib/design-system-v2/components";
 import { MdOutlineMoreHoriz } from "@react-icons/all-files/md/MdOutlineMoreHoriz";
 import { useApiClientContext } from "features/apiClient/contexts";
 import { NewRecordNameInput } from "../newRecordNameInput/NewRecordNameInput";
+import { upsertApiRecord } from "backend/apiClient";
+import { useSelector } from "react-redux";
+import { getUserAuthDetails } from "store/slices/global/user/selectors";
+import { getCurrentlyActiveWorkspace } from "store/features/teams/selectors";
+import { toast } from "utils/Toast";
 
 interface Props {
   record: RQAPI.ApiRecord;
 }
 
 export const RequestRow: React.FC<Props> = ({ record }) => {
   const [isEditMode, setIsEditMode] = useState(false);
-  const { updateRecordToBeDeleted, setIsDeleteModalOpen } = useApiClientContext();
+  const { updateRecordToBeDeleted, setIsDeleteModalOpen, onSaveRecord } = useApiClientContext();
+  const user = useSelector(getUserAuthDetails);
+  const teamId = useSelector(getCurrentlyActiveWorkspace);
+
+  const handleDuplicateRequest = useCallback(
+    async (record: RQAPI.ApiRecord) => {
+      const newRecord = {
+        ...record,
+        name: `${record.name} (Copy)`,
+      };
+      delete newRecord.id;
+      const result = await upsertApiRecord(user?.details?.profile?.uid, newRecord, teamId);
+      if (result.success) {
+        onSaveRecord(result.data);
+        toast.success("Request duplicated successfully");
+      }
+    },
```

Comment:
handle errors from `upsertApiRecord`

```suggestion
      return upsertApiRecord(user?.details?.profile?.uid, newRecord, teamId)
        .then((result) => {
          if (!result.success) {
            throw new Error("Failed to duplicate request");
          }
          onSaveRecord(result.data);
          toast.success("Request duplicated successfully");
        })
        .catch((error) => {
          console.error("Error duplicating request:", error);
          toast.error(error.message || "Unexpected error. Please contact support.");
        });```